sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ExpressionEvalHelper.scala

```diff
@@ -43,11 +43,38 @@ trait ExpressionEvalHelper extends GeneratorDrivenPropertyChecks {
 
   protected def checkEvaluation(
       expression: => Expression, expected: Any, inputRow: InternalRow = EmptyRow): Unit = {
-    val catalystValue = CatalystTypeConverters.convertToCatalyst(expected)
+    // No codegen version expects GenericArrayData
+    val catalystValue = expected match {
```

Comment:
As I added a comment, this change intends to give a primitive array to pass `Array(_, containsNull = true)` as a type to generate better code by using `UnsafeArrayData.fromPrimitiveArray`.
I agree to move `convertToCatalystUnsafe` to `ExpressionEvalHelper.scala`.